deploy/test/test_cases/TEST_ID_25_helm_default_retry_successful.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+set -euxo pipefail
+
+# This test verifies that default values for retry mechanism work
+setup_helm_environment
+
+pushd ../../
+  # Install HELM with a K8s Secret that does not exist
+  export LABELS="app: test-helm"
+  export DEBUG="true"
+  # A parameter that will force a failure
+  export CONJUR_APPLIANCE_URL="https://conjur-follower.${CONJUR_NAMESPACE_NAME}.svc.cluster.local/xyz"
+  fill_helm_chart_no_override_defaults
+  helm install -f "../helm/secrets-provider/ci/take-defaults-test-values.yaml" \
+    secrets-provider ../helm/secrets-provider \
+    --set-file environment.conjur.sslCertificate.value="test/test_cases/conjur.pem"
+popd
+
+pod_name=$($cli_with_timeout get pods --namespace=$APP_NAMESPACE_NAME --selector app=test-helm --no-headers | awk '{print $1}' )
+# Find initial authentication error that should trigger the retry
+$cli_with_timeout "logs $pod_name | grep 'CSPFK010E Failed to authenticate'"
```

Comment:
Solving this  [bug](https://github.com/cyberark/conjur-authn-k8s-client/issues/146) bug will change this test, right?

Reply:
Nope because I am intentionally failing the secrets provider

```
  # A parameter that will force a failure
  export CONJUR_APPLIANCE_URL="https://conjur-follower.${CONJUR_NAMESPACE_NAME}.svc.cluster.local/xyz"
```
`xyz` should be `api` but I am simulating a scenario when there is failure in authentication to trigger a retry